src/main/java/hudson/remoting/Engine.java

```diff
@@ -187,6 +188,9 @@ public Thread newThread(@NonNull final Runnable r) {
 
     private boolean disableHttpsCertValidation = false;
 
+    @CheckForNull
+    private HostnameVerifier hostnameVerifier;
```

Comment:
Following a general paradigm where everytime a boolean field exists to disable validation, a `HostNameVerifier` field exists beneath it. Following this convention consistently in all classes that do this allows for certain common utilities like `JnlpAgentEndpointResolver#openURLConnection` to be simplified.